### PR TITLE
enable range query on fast field for u64 compatible types

### PR DESCRIPTION
**Pull request by PSeitz**

### CODE REVIEW

File: src/query/range_query/range_query_u64_fastfield.rs [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+
+/// `FastFieldRangeWeight` uses the fast field to execute range queries.
+pub struct FastFieldRangeWeight {
+    field: Field,
+    left_bound: Bound<u64>,
+    right_bound: Bound<u64>,
+}
+
+impl FastFieldRangeWeight {
+    pub fn new(field: Field, left_bound: &Bound<u64>, right_bound: &Bound<u64>) -> Self {
```
Comment (fulmicoton) [inline src/query/range_query/range_query_u64_fastfield.rs]:
```suggestion
    pub fn new(field: Field, left_bound: Bound<u64>, right_bound: Bound<u64>) -> Self {
```

File: src/query/range_query/range_query.rs [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 use crate::query::explanation::does_not_match;
 use crate::query::range_query::range_query_ip_fastfield::IPFastFieldRangeWeight;
 use crate::query::{BitSetDocSet, ConstScorer, EnableScoring, Explanation, Query, Scorer, Weight};
 use crate::schema::{Field, IndexRecordOption, Term, Type};
 use crate::termdict::{TermDictionary, TermStreamer};
-use crate::{DocId, Score};
+use crate::{DateTime, DocId, Score};
 
 pub(crate) fn map_bound<TFrom, TTo, Transform: Fn(&TFrom) -> TTo>(
     bound: &Bound<TFrom>,
```
Comment (fulmicoton) [inline src/query/range_query/range_query.rs:17]:
```suggestion
    bound: Bound<TFrom>,
```

**Review by fulmicoton — APPROVED** [review]

File: src/query/range_query/range_query_u64_fastfield.rs [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+        }
+    }
+}
+
+impl Weight for FastFieldRangeWeight {
+    fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
+        let field_type = reader.schema().get_field_entry(self.field).field_type();
+        match field_type.fastfield_cardinality().unwrap() {
+            Cardinality::SingleValue => {
+                let ip_addr_fast_field = reader.fast_fields().u64_lenient(self.field)?;
```
Comment (shikhar) [inline src/query/range_query/range_query_u64_fastfield.rs]: The naming in this more generic (any u64 fast field) range query should not reference `ip_addr` right?

Comment (PSeitz) [reply src/query/range_query/range_query_u64_fastfield.rs]: yep, fixed

**Pull request merged**